### PR TITLE
Basic Python 3 support for webserver and db packages

### PR DESCRIPTION
**Pull request by gentlecat**
http://tickets.musicbrainz.org/browse/AB-60

### CODE REVIEW

File: webserver/utils.py [hunk 1]
```diff
@@ -4,6 +4,6 @@
 
 def generate_string(length):
     """Generates random string with a specified length."""
-    return ''.join([random.choice(string.ascii_letters.decode('ascii')
-                                  + string.digits.decode('ascii'))
-                    for _ in xrange(length)])
+    return ''.join([random.SystemRandom().choice(
```
Comment (gentlecat) [inline webserver/utils.py]: http://stackoverflow.com/a/23728630/272770

File: db/data.py [hunk 2]
```diff
@@ -62,12 +63,11 @@ def sanity_check_data(data):
...
-    for tag in tags.keys():
-        tag = tag.lower()
-        if tag not in _whitelist_tags:
-            del tags[tag]
-    data["metadata"]["tags"] = tags
+    clean_tags = copy.deepcopy(data["metadata"]["tags"])
+    for tag in data["metadata"]["tags"].keys():
+        if tag.lower() not in _whitelist_tags:
+            del clean_tags[tag]
+    data["metadata"]["tags"] = clean_tags
```
Comment (gentlecat) [inline db/data.py]: http://chatlogs.musicbrainz.org/musicbrainz-devel/2015/2015-07/2015-07-31.html#T17-46-54-91631

File: webserver/views/test/test_data.py [hunk 3]
```diff
@@ -13,7 +13,7 @@ def setUp(self):
 
     def test_api(self):
         resp = self.client.get(url_for('data.api'))
-        self.assertEquals(resp.status_code, 302)  # Should redirect to data page
```
Comment (gentlecat) [inline webserver/views/test/test_data.py]: `assertEquals` is deprecated.

**Comment by alastair** [comment]
:+1:

**Pull request merged**